HackAtUCI/irvinehacks-site
# Restructure first/last name in user records

Resolves #422. Can look into simplify actual participants query later on.

#### Changes
- Move `user_record.py` from `utils` to `models`
- Place first/last name on root of user record for all participants
  - Previously, applicants had their name stored in `application_data`
  - The new structure simplifies how all participants are queried
  - Anybody assigned a role should have a name
- Reorganize ApplicationData models and create new `BareApplicant` model
  - Constrain some database queries by using explicit fields
  - This simplifies some of the unit tests not needing application data
- Update data properties for applicants referenced in Admin site

#### Testing
If you have any old user records in your local database, you will need to update them with the new structure.
1. Impersonate an applicant and submit an application (adjust deadline if needed)
2. Create a user record with an admin role and impersonate that user
3. On the Admin site, observe the applicant is listed properly in the Applicants page and the details for the individual applicant also load properly
4. Manually update the applicant's status to `ATTENDING`
5. Observe the applicant name is listed properly in the Participants page on the Admin site

The waiver submission handler was not tested manually, but test coverage there is pretty good. Other admin-related functionality involving emails and personalizations was not tested (still need to add unit tests).
